lib/nats/jetstream/api.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require_relative "api/dsl"
+require_relative "api/group"
+require_relative "api/endpoint"
+require_relative "api/request"
+require_relative "api/response"
+
+module NATS
+  class JetStream
+    class Api
```

Comment:
How about uppercase name here?
```suggestion
    class API
```

Reply:
That's a good idea 👍

I initially used `Api` name because the legacy API already declares `API` constant, and I was afraid it clashed with the simplified API. 

Looking over it, it seems that the only point is `StreamCreateResponse`, which exists in both APIs. I renamed `StreamCreateResponse` into `StreamCreate` for the legacy API; I believe it should not create any problems for the existing user code.